### ISSUE TITLE
Bug: fail to display: separate (different, independent) contacts which share same/overlapping phone#s...

### ISSUE THREAD

**Issue by jwmh** [SimpleMobileTools/Simple-Contacts#629]
For example:
Three people work at the same company, Alice, Bob, Charlie
.
They have different *extensions* at work,
  but the same original 10digit telephone number.
.
Sample:

Alice
Work Phone: (555)555-5555,,,001#

Bob
Work phone: (555)555-5555,,,002#

Charlie
Work phone: (555)555-5555,,,003#

The above examples illustrate how I like to use my Contacts directory.

Another example is calling different departments within a large corporate, or goverment, organization:

I need to:
- call the main#,
- select 1 for english,
- enter my state medicaid ID# (a nine-digit string),
- press the pound or hash sign,

And then....
If i am calling tech support (because their web portal has bugs),
Then i press 2,
Then i press 7 to speak with someone

Or...
If i am calling to change my health network of providers,
I instead press 3,

Or...
If i am calling to update my contact info,
I instead press 4,
Then i press 1

Example:
Oregon Health Plan - tech support:
888-888-8888,,1,,000000000#,,,2,,2

Oregon Health Plan -- change address/contact info:
888-888-8888,,1,,000000000#,,,4,,1

~~~~

#####

*****

TL:DR;
These dont show up properly as separate contacts.

Something about how Simple Contacts **parses*** 
and stores, assigns in a database...

... Something about the algorithm seems inadequate for differentiating / telling the difference  between these use cases.
